docs/types.md

```diff
@@ -0,0 +1,47 @@
+# Using Types
+
+When you have a `ReflectionParameter`, you can determine types in the following
+ways:
+
+```php
+class MyClass
+{
+    /**
+     * @param array $myMethod
+     */
+    public function myMethod(array $myMethod = [])
+    {
+        // ... stuff ...
+    }
+}
+```
+
+```php
+
+$class = $reflector->reflect('MyClass');
+$method = $class->getMethod('myMethod');
+$parameter = $method->getParameter('myParameter');
+
+echo $parameter->getTypeHint(); // Will fetch a Type object for the language hint
+var_dump($parameter->getTypes()); // Will fetch an array of Type objects for the docblock hint
```

Comment:
I think the method should be named `getDocBlockTypes` or such to avoid any possible confusion